test/unit/search/search_scheme_algorithm_test.cpp

```diff
@@ -0,0 +1,339 @@
+                                std::vector<uint8_t> const & error_distribution, time_t const seed,
+                                auto const & blocks_length, auto const & ordered_blocks_length,
+                                uint64_t const start_pos)
+{
+    using char_t = typename text_t::value_type;
+
+    uint64_t const pos = std::rand() % (text.size() - query_length + 1);
+    text_t const orig_query = text | ranges::view::slice(pos, pos + query_length);
+
+    // Modify query s.t. it has errors matching error_distribution.
+    auto query = orig_query;
+    uint64_t current_blocks_length = 0;
+    for (uint8_t block = 0; block < search.blocks(); ++block)
+    {
+        uint64_t const single_block_length = ordered_blocks_length[block];
+        EXPECT_LE(error_distribution[block], single_block_length);
+        if (error_distribution[block] > single_block_length)
+        {
+            debug_stream << "Error in block " << block << "(+ 1): " << error_distribution[block]
+                         << " errors cannot fit into a block of length " << single_block_length << "." << '\n'
+                         << "Error Distribution: " << error_distribution << '\n';
+            exit(1);
+        }
+
+        // Choose random positions in the query sequence for substitutions. Repeat until all error positions are unique.
+        std::vector<uint8_t> error_positions(error_distribution[block]);
+        do
+        {
+            error_positions.clear();
+            for (uint8_t error = 0; error < error_distribution[block]; ++error)
+                error_positions.push_back(std::rand() % single_block_length);
+            std::sort(error_positions.begin(), error_positions.end());
+        } while (std::adjacent_find(error_positions.begin(), error_positions.end()) != error_positions.end());
+
+        // Construct query sequence with chosen error positions.
+        for (uint8_t error = 0; error < error_positions.size(); ++error)
+        {
+            uint64_t const pos = error_positions[error] + current_blocks_length;
+            // Decrease alphabet size by one because we don't want to replace query[pos], with the same character.
+            uint8_t new_rank = std::rand() % (alphabet_size_v<char_t> - 1);
+            // If it is a match now, it can't be the highest rank of the alphabet. Thus we set it to the highest rank.
+            if (new_rank == to_rank(query[pos]))
+                new_rank = alphabet_size_v<char_t> - 1;
+            assign_rank(query[pos], new_rank);
+        }
+        current_blocks_length += single_block_length;
+    }
+
+    std::vector<uint64_t> hits_trivial, hits_ss;
+
+    auto delegate_trivial = [&hits_trivial](auto const & it)
+    {
+        auto const & hits_tmp = it.locate();
+        hits_trivial.insert(hits_trivial.end(), hits_tmp.begin(), hits_tmp.end());
+    };
+
+    auto delegate_ss = [&hits_ss](auto const & it)
```

Comment:
we have a space between lambda capture and parameters, this affects a lot of places, but can easily be fixed...